crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -75,13 +75,19 @@ def current_step_state
       progress[:steps][current_step] || {}
     end
 
-    def start_step
+    # 'step' is name of the step user wants to start.
+    # If `out_of_order` is true, the step could be executed out of order
+    def start_step(step_name, out_of_order = false)
```

Comment:
Well, it depends on the role we assign to this library. If it's supposed to be a simple wrapper around the status file then no, such a mapping doesn't belong there. On the other hand, maybe we should put some more functionality in the library to make it more convenient for the client code?